packages/theme/src/Provider.tsx

```diff
@@ -98,6 +103,30 @@ const Provider: React.FC<React.PropsWithChildren<ProviderProps>> = ({
     [currentTheme, deviceWidth, breakpoints]
   );
 
+  React.useEffect(() => {
+    const listener = Dimensions.addEventListener("change", ({ window }) =>
+      setDeviceWidth(window.width)
+    );
+    return () => {
+      listener.remove();
+    };
+  });
+
+  React.useEffect(() => {
+    const run = async () => {
```

Comment:
Consider if the UI shakes slightly during initialization due to the absence of a wait state when loading the theme.

👉  Example: the UI may briefly display with theme1 before switching to theme2, causing a flicker.

If thorough testing shows no issue, you can disregard this comment.